prow/flagutil/github.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flagutil
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"net/url"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/git"
+	"k8s.io/test-infra/prow/github"
+)
+
+// GitHubOptions holds options for interacting with GitHub.
+type GitHubOptions struct {
+	endpoint            Strings
+	TokenPath           string
+	deprecatedTokenFile string
+}
+
+// AddFlags injects GitHub options into the given FlagSet.
+func (o *GitHubOptions) AddFlags(fs *flag.FlagSet) {
+	o.endpoint = NewStrings("https://api.github.com")
+	fs.Var(&o.endpoint, "github-endpoint", "GitHub's API endpoint (may differ for enterprise).")
+	fs.StringVar(&o.TokenPath, "github-token-path", "/etc/github/oauth", "Path to the file containing the GitHub OAuth secret.")
+	fs.StringVar(&o.deprecatedTokenFile, "github-token-file", "", "DEPRECATED: use -github-token-path instead.  -github-token-file may be removed anytime after 2019-01-01.")
+}
+
+// Validate validates GitHub options.
+func (o *GitHubOptions) Validate(dryRun bool) error {
+	for _, uri := range o.endpoint.Strings() {
+		if _, err := url.ParseRequestURI(uri); err != nil {
+			return fmt.Errorf("invalid -github-endpoint URI: %q", uri)
+		}
+	}
+
+	if o.deprecatedTokenFile != "" {
```

Comment:
hummmmmmm we have plank deployment doesn't need github so we set `--github-token-file=`, and this skipped that check :-\

Reply:
> hummmmmmm we have plank deployment doesn't need github so we set --github-token-file=, and this skipped that check :-\

I can set an unlikely sentinel instead of using the empty string to detect folks setting explicitly to the empty string, although we'll have to figure out [error handling][1] to know what to do with empty strings.

[1]: https://github.com/kubernetes/test-infra/pull/9503/#pullrequestreview-168144833